src/DrupalCI/Configuration/EnvironmentVariables.php

```diff
@@ -0,0 +1,381 @@
+<?php
+
+namespace DrupalCI\Configuration;
+
+use Drupal\Component\Annotation\Variable;
+use DrupalCI\Injectable;
+use Pimple\Container;
+
+/**
+ * Object used for mapping environment variables with local variables.
+ */
+class EnvironmentVariables implements Injectable {
+
```

Comment:
Like we chatted over slack, we're trying to keep everything about a plugin specific to that plugin, inside the plugin class, so the properties and special parsers/processors for those should live in the plugins, not in a central monolithic class.